fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -3720,20 +3721,25 @@ private Interaction buildInteractionStatement(TypeRestfulInteraction value) {
      */
     private Bundle createSearchBundle(List<Resource> resources, FHIRSearchContext searchContext)
         throws FHIROperationException {
+        
+        // throws if we have a count of more than 2,147,483,647 resources
+        UnsignedInt totalCount = com.ibm.fhir.model.type.UnsignedInt.of(Math.toIntExact(searchContext.getTotalCount()));
```

Comment:
I like this check, even thought I would believe the DB server query will most likely timeout if there are such huge number of matched resource ...